core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java

```diff
@@ -259,7 +304,7 @@ public synchronized void lockAsync(final AsyncLockWatcher lw, byte[] data) {
     lockWasAcquired = false;
 
     try {
-      final String asyncLockPath = zooKeeper.putEphemeralSequential(path + "/" + LOCK_PREFIX, data);
+      final String asyncLockPath = zooKeeper.putEphemeralSequential(path + "/" + THIS_ZLOCK, data);
```

Comment:
In the case where the ephemeral node is created on the server and that is not successfully reported back, will a second ephemeral node be created?  For example could we end up with two ephemeral nodes w/ the same UUID like: 

```
zlock#44755fbe-1c9e-40b3-8458-03abaf950d7e#0000000002
zlock#44755fbe-1c9e-40b3-8458-03abaf950d7e#0000000003
```

If so, would the 2nd one created be returned to this code and would the 1st one continue to exists?

Reply:
I think the idea would be to avoid creating a second node. From what I understand, an attempt would be made to recover the (created but not returned) node via its UUID. A search among the parent paths children could take place when this error occurs and if a node is found with the UUID then its path would be returned for use. I outlined this [here](https://github.com/apache/accumulo/issues/637#issuecomment-756902064). I think this recovery process would probably make sense to be within the retry logic in putEphemeralSequential where the exception could be caught to begin the recovery attempt.

Reply:
If it can occur, one option would be to change how putEphemeralSequential works.  Another option is to handle multiple nodes w/ the same UUID in the child list when deciding if the lock is held.  If every child less than the node id returned by putEphemeralSequential has the same UUID, then we know we have the lock.

Reply:
Another option may be to leave putEphemeralSequential as is and always get the children list after calling putEphemeralSequential and delete anything w/ the same UUID that is not equal to what putEphemeralSequential returned.

Reply:
I think if this error occurs where the node was created but not reported, putEphemeralSequential would not return anything.

Reply:
resolving this, we can pick it up below.